Userland/Libraries/LibLocale/CMakeLists.txt

```diff
@@ -3,12 +3,8 @@ include(${SerenityOS_SOURCE_DIR}/Meta/CMake/locale_data.cmake)
 if (DEFINED LOCALE_DATA_SOURCES)
     set(SOURCES ${LOCALE_DATA_SOURCES})
     set(GENERATED_SOURCES ${CURRENT_LIB_GENERATED})
-    if (SERENITYOS)
-        serenity_lib(LibLocaleData localedata)
-    else()
-        add_library(LibLocaleData OBJECT ${SOURCES})
-        serenity_generated_sources(LibLocaleData)
-    endif()
+    add_library(LibLocaleData OBJECT ${SOURCES})
+    serenity_generated_sources(LibLocaleData)
```

Comment:
I think these 2 lines can now be accomplished with:
```cmake
serenity_lib(LibLocaleData localedata TYPE OBJECT)
```
(not a big deal if Ladybird and everything else links and runs fine, I just remember hitting some linker issue last time I manually added an object lib without invoking `serenity_lib`)